source/graphql/migrate-hasura.txt

```diff
@@ -0,0 +1,263 @@
+
+Migrate your MongoDB Atlas GraphQL schema to Hasura's GraphQL schema. This
+involves defining your types, queries, mutations, and subscriptions in the
+Hasura Console or the Hasura CLI.
+
+.. procedure::
+
+   .. step::
+
+      In the Hasura console click the Data tab and select your MongoDB database
+      from the left hand pane.
+
+      You'll see all the collections from your MongoDB database.
+
+      .. image:: /images/migration/collections.png
+         :alt: Atlas collections are listed in a table.
+
+   .. step::
+
+      To generate a schema for a collection, click on the :guilabel:`Track`
+      button next to the desired collection.
+
+      You have three options to infer the schema of a collection:
+
+      - Infer the schema from a sample document you copy and paste from your
+        collection.
+      - :guilabel:`Use Database Schema` allows you to use your database's
+        validation schema if one is present in your database (a GraphQL schema
+        will be automatically generated for you).
+      - :guilabel:`Use Existing Logical Model` allows you to select a previously
+        created logical model.
+
+      .. image:: /images/migration/generate-schema.png
+         :alt: Generate a new schema by using a sample document.
+
+   .. step::
+
+      Click the :guilabel:`Validate` button to validate the JSON document. In
+      the next step, you will see the models derived from this document.
+      Finally, click the :guilabel:`Track Collection` button.
+
+Test Your GraphQL Queries
+-------------------------
+
+After tracking a collection and getting a schema, you can navigate to the API
+Explorer page on the Hasura Console to test out some GraphQL queries.
+
+Hasura also uses the GraphiQL interface, which is similar to how you test
+queries in Atlas App Services.
+
+.. image:: /images/migration/test-queries.png
+   :alt: Use the GraphiQL interface to test queries.
+
+Authorization and Authentication
+--------------------------------
+
+Hasura does not directly handle authentication. Instead, it relies on session
+variables provided by an external authentication service. These session
+variables include user, role, and organization information crucial for
+determining data access rights. For details, refer to the `Hasura authentication
+docs <https://hasura.io/docs/latest/auth/overview/>`_
+
+Data access permissions, including roles and rule expressions, can be converted
+into Hasura role-based permission rules.
+
+All Authentication methods that Atlas provides are compatible with Hasura's
+Webhook and JWT auth methods. If you are using Email/Pass, Anonymous, or API Key
+authentication, use Hasura Webhook. If you are using a Custom JWT, integrate
+directly with Hasura's JWT auth method.
+
+It is recommended to use an external IdP for managing authentication
+processes. For enhanced security and flexibility, organizations typically manage
+tokens and keys centrally rather than adjacent to the database.
```

Comment:
Nit: this particular comment is a little strange to read in the App Services docs, where we talk about how we can manage auth, and leverage our auth services in the SDK to provide a fully-integrated solution.

I might remove this paragraph, and add the first sentence to the paragraph below - something like:

> Hasura recommends using an external IdP for managing authentication processes. You can integrate Hasura with...

Reply:
I agree that the wording comes off a bit jarring. This statement came from an eng request to specify in the guides that GraphQL users should also migrate their app users/auth to an external IdP to avoid us becoming an auth provider as a service (i.e. we don’t want to be supporting these customers going forward if they were only using App Services for GraphQL).
Can we change this to:
> Hasura recommends using an external IdP for managing authentication processes for enhanced security and flexibility. You can integrate Hasura with...

Reply:
Great point. Thank you! I've updated.